### PR TITLE
Add upload artifact step for PR level

### PR DESCRIPTION
**Pull request by spirosoik**
#### Summary
Add an action to upload the artifact to an S3 bucket. In parallel add:
- license
- codeowners
- readme
- 
#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-4681

### CODE REVIEW

File: .github/workflows/plugin-ci.yml [hunk 1]
```diff
@@ -81,7 +81,25 @@ jobs:
           cache: true
 
       - name: ci/setup
-        uses: spirosoik/actions/plugin-ci/setup@main
+        uses: mattermost/actions/plugin-ci/setup@main
 
       - name: ci/test
```
Comment (phoinixgrr) [inline .github/workflows/plugin-ci.yml:86]: Should this be renamed to ci/build?

File: .github/workflows/plugin-ci.yml [hunk 2]
```diff
@@ -81,7 +81,25 @@ jobs:
...
-        uses: spirosoik/actions/plugin-ci/build@main
+        uses: mattermost/actions/plugin-ci/build@main
```
Comment (phoinixgrr) [inline .github/workflows/plugin-ci.yml:87]: Should we SHA hash pin this action ?

File: .github/workflows/plugin-ci.yml [hunk 3]
```diff
@@ -81,7 +81,25 @@ jobs:
           cache: true
 
       - name: ci/setup
-        uses: spirosoik/actions/plugin-ci/setup@main
+        uses: mattermost/actions/plugin-ci/setup@main
```
Comment (phoinixgrr) [inline .github/workflows/plugin-ci.yml:84]: Should we SHA hash pin this action ?

File: .github/workflows/plugin-ci.yml [hunk 4]
```diff
@@ -81,7 +81,25 @@ jobs:
...
+        uses: actions/download-artifact@9782bd6a9848b53b110e712e20e42d89988822b7 # v3.0.1
+        with:
+          name: dist
+          path: dist
+
+      - name: ci/prepare-artifact
+        run: mv dist/*.tar.gz dist/$GITHUB_REPOSITORY-ci.tar.gz
+
+      - name: ci/upload-artifact
+        uses: mattermost/actions/artifact-upload@main
```
Comment (phoinixgrr) [inline .github/workflows/plugin-ci.yml:102]: Should we SHA hash pin this action ?

**Comment by spirosoik** [comment]
@phoinixgrr good point.

**Comment by spirosoik** [comment]
@phoinixgrr could you please review again?

**Review by phoinixgrr — APPROVED** [review]
:shipit:

**Pull request merged**